network.go

```diff
@@ -195,6 +195,12 @@ func (n *network) CreateEndpoint(name string, options ...EndpointOption) (Endpoi
 	if name == "" {
 		return nil, ErrInvalidName(name)
 	}
+
+	e, _ := n.EndpointByName(name)
```

Comment:
I thought abt that... but if we check for err.. then the code will look like this -
```
_, err := n.EndpointBytName(name)
if err == nil {
    return nil, types.ForbiddenErrorf("service endpoint with name %s already exists", name)
}
```
We will have to check for `err == nil`, which I think is equivalent :-)